### PR TITLE
Implementation of parallelization to `MDAnalysis.analysis.dssp`

### LINKED ISSUE #4674: `MDAnalysis.analysis.dssp`: Implement parallelization or mark as unparallelizable
Given a newly implemented parallelization (https://docs.mdanalysis.org/dev/documentation_pages/analysis/parallelization.html), implement it for all classes in `analysis.dssp` module. You may use `analysis.RMSD` in `develop` as a reference.

You need to add:

 - appropriate implementation of result aggregation for AnalysisBase subclasses in `analysis.dssp` module (or explicitly limit available backends if the classes aren't parallelizable)
 - a boilerplate fixture(s) to `testsuite/analysis/conftest.py`, analogous with existing ones
 - a `client_...` fixtures to all tests using in `testsuite/MDAnalysisTests/analysis/test_dssp.py`, and modify the way run() method is called

### PR DESCRIPTION
**Pull request by talagayev**
Fixes #4674 

Changes made in this Pull Request:

 - Parallelization of the backend support to the class `DSSP` in `dssp.py`
 - Addition of `versionchanged` to mention the addition of the parallelization
 - Addition of parallelization tests in `test_dssp.py` and fixtures in `conftest.py`
 - Updated Changelog

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4721.org.readthedocs.build/en/4721/

### CODE REVIEW

File: package/MDAnalysis/analysis/dssp/dssp.py [hunk 1]
```diff
@@ -276,8 +276,24 @@ class DSSP(AnalysisBase):
 
 
     .. versionadded:: 2.8.0
+
+    .. versionchanged:: 2.8.0
```
Comment (yuxuanzhuang) [inline package/MDAnalysis/analysis/dssp/dssp.py]: I don't think you need this as it was created in 2.8.0 (l278)

Comment (yuxuanzhuang) [inline package/MDAnalysis/analysis/dssp/dssp.py]:
```suggestion
```

**Review by yuxuanzhuang — APPROVED** [review]
lgtm!

File: package/MDAnalysis/analysis/dssp/dssp.py (hunk 1 above)
Comment (talagayev) [reply package/MDAnalysis/analysis/dssp/dssp.py]: Ah okay, I wasn't sure so I added it to be save :)

**Comment by talagayev** [comment]
I would have then also have a quick question:
for example when I try to add the parallelization to `desnity.py` for the `DensityAnalysisI` get this during the `test_density.py` pytests as a Failure message for `test_userdefn_boxshape:`

`AttributeError: 'DensityAnalysis' object has no attribute '_grid'`

I am not sure how to fix such cases, should I just create a PR so that the reviewers can take a look at it and suggest a fix?

**Comment by RMeli** [comment]
@talagayev it would help to see the full backtrace for the error, otherwise it's a bit hard to help.

> should I just create a PR so that the reviewers can take a look at it and suggest a fix?

This might be useful if you are planning to work on it regardless. You can open a PR as "draft", to make it explicitly it is work in progress but want early feedback/need some help/want to test in CI.

**Comment by talagayev** [comment]
> @talagayev it would help to see the full backtrace for the error, otherwise it's a bit hard to help.
> 
> > should I just create a PR so that the reviewers can take a look at it and suggest a fix?
> 
> This might be useful if you are planning to work on it regardless. You can open a PR as "draft", to make it explicitly it is work in progress but want early feedback/need some help/want to test in CI.

hey @RMeli, perfect that is what i was looking for, thanks :) I will then create the PR as a draft to get some feedback, since there are also some other cases like `MDAnalysis.analysis.align` where I tried to parallelize them and think that they are not possible to parallelize, but I can then also create a draft PR with the Failure message, so that I get feedback if I am correct that they can not be parallelized and then adjust everything to mark them as unparallelizable :)

**Comment by orbeckst** [comment]
@talagayev can you please resolve conflicts?

@yuxuanzhuang can you please squash-merge when you're happy (please also write a good log messages — see, for example, PR #4718 )

**Comment by talagayev** [comment]
@orbeckst @yuxuanzhuang sorry for the late replies, I am on holiday currently till mid of next week and thus have only sometimes access to the laptop 🙈 

@yuxuanzhuang should I do anything else or now that you merged the development branch into the PR branch it is ready to be merged? :)

**Comment by yuxuanzhuang** [comment]
@talagayev I’ll take care of it, no worries! You’ve been incredibly helpful in moving this forward, no need to apologize!

**Comment by talagayev** [comment]
@yuxuanzhuang Nice, thanks to you too for looking into the PR's and helping with them :)

I would then create the couple of PR's as drafts today for some functions where I am not sure  if they can be parallelized so that you and the other core devs can take a look into it and give your opinions, since I have them prepared locally and am at the laptop right now.

Happy to help :)

**Pull request merged**